src/libs/ajax/loaded-data/useLoadedData.ts

```diff
@@ -1,7 +1,23 @@
 import { useState } from 'react'
-import LoadedState from 'src/libs/type-utils/LoadedState'
+import LoadedState, { ErrorState } from 'src/libs/type-utils/LoadedState'
 import { isFetchResponse } from 'src/libs/type-utils/type-helpers'
 
+
+export interface UseLoadedDataArgs<T> {
+  /**
+   * An optional handler that will be called if there is an error.
+   * Note that LoadedData object typing already allows expression of error status, convenient for consumption by
+   * visual components.  This handler is to accommodate additional side effects within the hook consuming
+   * useLoadedData hook
+   * @param state - the error state as of when the error happened
+   * @example
+   * const [pendingCreate, setPendingCreate] = useLoadedData<true>({
+   *   onError: (errState) => ReportError(errState.error)
+   * })
+   */
+  onError?: (state: ErrorState<T, unknown>) => void
```

Comment:
One last idea, it occurs to me that if the default parameter list had a no-op operation for `onError` then you no longer need to make it optional and wouldn't need to check for it below. e.g.
```
  onError: (state: ErrorState<T, unknown>) => void

[...]

export const useLoadedData = <T>(hookArgs: UseLoadedDataArgs<T> = { onError = () => {} }): UseLoadedDataResult<T> => {

[...]

      hookArgs.onError(errorResult)

```
Unless there's a case where you'd want to allow the caller to explicitly pass it in as `undefined`.

Reply:
hmmm...  a good thought, but feels a little heavy-handed just to avoid a bit of internal handling.

The spirit of going with a single args object as opposed to a growing list of function arguments is for a number of longterm benefits:
- argument "names" are auto-documented since they must be given as part of the arg object by the consumer, making code more readable.
- argument order no longer matters
- argument optional/required no longer conflicts with argument list order
- a number of object conveniences, including TS.

And long-term what I tend to do with larger argument objects is to have a default object with the default if any, and merge the incoming arg object over the default one at the top of the function/class.  But here I am keeping things simple since we may never need more than this one arg.

Reply:
This approach is still using a single args object. The only difference is that the single args object default value isn't empty anymore. So all the other attributes you mentioned are still true.